instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingProcessingContext.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2013-2024 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.kafka.streams;
+
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContext.Injector;
+import java.io.File;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.StreamsMetrics;
+import org.apache.kafka.streams.processor.Cancellable;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.processor.Punctuator;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.TaskId;
+import org.apache.kafka.streams.processor.api.ProcessingContext;
+import org.apache.kafka.streams.processor.api.RecordMetadata;
+
+class TracingProcessingContext<C extends ProcessingContext> implements ProcessingContext {
```

Comment:
Super  nit, just curious if we could mark this class `abstract`, it seems to be a building block but not used by itself

Reply:
good idea. thanks!